pydantic/main.py

```diff
@@ -243,9 +243,9 @@ def __setattr__(self, name, value):
         elif self.model_config['frozen']:
             raise TypeError(f'"{self.__class__.__name__}" is frozen and does not support item assignment')
         elif self.model_config['validate_assignment']:
-            values, fields_set = self.__pydantic_validator__.validate_assignment(name, value, self.__dict__)
+            values = self.__pydantic_validator__.validate_assignment(self.__dict__, name, value)
             _object_setattr(self, '__dict__', values)
-            self.__fields_set__ |= fields_set
+            self.__fields_set__ |= {name}
```

Comment:
I don't know if this is the desired logic here now that `validate_assignment` no longer returns `fields_set`.

I will say the current set of unit tests pass with or without this line; not sure if that's expected due to changes in the logic in `validate_assignment`...